examples/typescript/pipe-screen-time-storyteller/pipe.ts

```diff
@@ -0,0 +1,353 @@
+}
+
+type ContentType = "ocr" | "audio" | "all"; // Add this near the top of your file
+
+interface ContentItem {
+    // Define the structure of your content items here
+    // For example:
+    id: string;
+    content: string;
+    timestamp: string;
+    // Add other relevant fields
+}
+
+async function getAIProvider(config: Config): Promise<AIProvider> {
+    return { provider: config.aiProvider, model: config.models[config.aiProvider] };
+}
+
+// async function queryScreenpipe(params: ScreenpipeQueryParams): Promise<ScreenpipeResponse | null> {
+//     try {
+//         console.log("Calling screenpipe", JSON.stringify(params));
+//         const result = await pipe.queryScreenpipe(params);
+//         console.log("Got", result.data.length, "items from screenpipe");
+//         return result;
+//     } catch (error) {
+//         console.error("Error querying screenpipe:", error);
+//         return null;
+//     }
+// }
+
+async function generateNarrativeSummary(screenData: ContentItem[], provider: AIProvider, config: Config): Promise<NarrativeSummary> {
+    // Limit the number of items we send to the AI
+    const maxItems = 100;
+    const truncatedData = screenData.slice(0, maxItems);
+
+    const prompt = `You're an AI companion living inside the user's devices. Based on the following sample of screen data (${truncatedData.length} out of ${screenData.length} total items), write a friendly, slightly sassy diary entry about what you've observed today:
+
+    ${JSON.stringify(truncatedData)}
+
+    Your diary entry should:
+    1. Comment on the user's habits
+    2. Make playful jokes about their app usage
+    3. Offer 3 pieces of advice for tomorrow
+    4. Use a mix of tech slang and emojis for a modern feel
+
+    Return the summary as a JSON object with the following structure:
+    {
+        "summary": "Your sassy AI companion diary entry",
+        "mood": "An emoji that represents the overall mood of the day",
+        "keyInsights": ["Array of 3 key insights or observations, each with an relevant emoji"]
+    }`;
+
+    let response: Response;
+    let result: any;
+
+    try {
+        if (provider.provider === "claude") {
+            response = await fetch("https://api.anthropic.com/v1/messages", {
+                method: 'POST',
+                headers: {
+                    "Content-Type": "application/json",
+                    "x-api-key": config.apiKeys.claude,
+                    "anthropic-version": "2023-06-01"
+                },
+                body: JSON.stringify({
+                    model: provider.model,
+                    messages: [{ role: "user", content: prompt }],
+                    max_tokens: 1000,
+                }),
+            });
+            result = await response.json();
+            console.log("Claude API response:", JSON.stringify(result, null, 2));
+            if (result.content && Array.isArray(result.content) && result.content[0] && result.content[0].text) {
```

Comment:
tip: avoid nesting if and try catch like this, it make the code very hard to read/change/understand

instead using early exit (or ask chatgpt how)